File: applications/summersim18/wrapped/python/add-showcase/right-add.py
from numb import *

if __name__ == '__main__':
    Numb.setup(cache_out='add-cache2', cache_in='add-cache1', precision=20, strategy='use-cache')
    a = Numb(0.001)
    b = Numb(1)
    c = Numb(-1)

    d = a + (b + c)
    print "0.001 + (1 + -1) = {0}".format(d)

File: applications/summersim18/wrapped/cpp/add-showcase/right-add.cpp
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include "../../../../../cpp/numb/numb.hpp"

int main()
{
    Numb::setup("add-cache2", "add-cache1", 20, "use-cache");
    Numb a(0.001);
    Numb b(1);
    Numb c(-1);

    Numb d = a + (b + c);
    cout << "0.001 + (1 + -1) = " << d << endl;

    return 0;
}
